src/lib.rs

```diff
@@ -14,26 +14,32 @@ use tree_sitter::Query;
 
 mod args;
 mod language;
+mod line_buffer;
```

Comment:
I didn't keep the existing `grep_*` crates as separate crates (nor even strictly in subdirectories like they were in), just "hoisted" all of the relevant modules to the top level